### ISSUE TITLE
GlideRecord Not looping

### ISSUE THREAD

**Issue by jbgriffith** [thewhitespace/Xplore#11]
I'm running Jakarta Patch5. 
I ran a simple query to delete some records, and noticed that it was too fast for what I wanted.

```javascript
var gr = new GlideRecord('sys_user_group');
gr.addEncodedQuery("sys_created_onON2017-11-22@javascript:gs.dateGenerate('2017-11-22','start')@javascript:gs.dateGenerate('2017-11-22','end')");

gr.query();
while (gr.next()) {
    gs.log(gr.u_friendly_name, 'Deleting Groups');
    gr.deleteRecord();
}
```

The result was: 
As you can see it took ~4 seconds, I have about 8000 of these groups that were created on the specified date. So I added the log to see how many records it was deleting, and it was one at a time... Since it's showing "true" as the Value, I'm assuming it's showing the next value returned from gr.next() (the condition in the while loop)
 

![Looping issue](https://ibin.co/3j72j0Jgghng.jpg)

I then decided to do another sanity check before I send you an issue.

```javascript
var count = 0;
var gr = new GlideRecord('sys_user_group');
gr.addEncodedQuery("sys_created_onON2017-11-22@javascript:gs.dateGenerate('2017-11-22','start')@javascript:gs.dateGenerate('2017-11-22','end')");

gr.query();
gs.addInfoMessage('# records: ' + gr.getRowCount());

while (gr.next()) {
    gs.log(gr.u_friendly_name, 'Deleting Groups');
    var wasDeleted = gr.deleteRecord();
    if (wasDeleted) { 
        count++;
    }
}
gs.addInfoMessage('# Deleted: ' + count);
```

![Looping issue 2](https://ibin.co/3j77ANeFGTZB.jpg)

**Comment by jbgriffith** [comment]
wait until I tell you to continue looking at this, I may have a different issue.

**Comment by jneale** [comment]
Hi James,
I've ran that type of query many times with no issue. You can also run ```gr.deleteMultiple()``` without the while loop to just batch delete them (obviously you can't add your own code as in the loop though).
I think what you've found is actually a known bug where boolean values are being shown in reverse of what they actually are - so while it's showing true, the chances are the ```gr.deleteRecord()``` result is actually false. I have fixed this in v4.5 which I'll try to release today.
Cheers,
James

**Comment by jneale** [comment]
v4.5 has been released with inverted boolean issue fixed. Closing issue.
